workspaces/diff-engine/src/events/spec_chunk.rs

```diff
@@ -0,0 +1,332 @@
+use super::{RfcEvent, SpecEvent};
+use cqrs_core::Event;
+use serde_json;
+use std::convert::TryFrom;
+
+#[derive(Debug)]
+pub enum SpecChunkEvent {
+  Root(RootChunkEvent),
+  Batch(BatchChunkEvent),
+  Unknown(UnknownChunkEvent),
+}
+
+#[derive(Debug)]
+pub struct RootChunkEvent {
+  pub id: String,
+  pub name: String,
+  pub events: Vec<SpecEvent>,
+}
+
+#[derive(Debug)]
+pub struct BatchChunkEvent {
+  pub id: String,
+  pub name: String,
+  pub parent_id: String,
+  pub events: Vec<SpecEvent>,
+}
+
+#[derive(Debug)]
+pub struct UnknownChunkEvent {
+  pub name: String,
+  pub events: Vec<SpecEvent>,
+}
+
+impl SpecChunkEvent {
+  pub fn into_events_iter(self) -> impl Iterator<Item = SpecEvent> {
+    let events = match self {
+      SpecChunkEvent::Root(chunk) => chunk.events,
+      SpecChunkEvent::Batch(chunk) => chunk.events,
+      SpecChunkEvent::Unknown(chunk) => chunk.events,
+    };
+
+    events.into_iter()
+  }
+
+  pub fn name(&self) -> String {
+    match self {
+      SpecChunkEvent::Root(chunk) => chunk.name.clone(),
+      SpecChunkEvent::Batch(chunk) => chunk.name.clone(),
+      SpecChunkEvent::Unknown(chunk) => chunk.name.clone(),
+    }
+  }
+
+  pub fn len(&self) -> usize {
+    match self {
+      SpecChunkEvent::Root(chunk) => chunk.events.len(),
+      SpecChunkEvent::Batch(chunk) => chunk.events.len(),
+      SpecChunkEvent::Unknown(chunk) => chunk.events.len(),
+    }
+  }
+}
+
+impl Event for SpecChunkEvent {
+  fn event_type(&self) -> &'static str {
+    match self {
+      SpecChunkEvent::Root(evt) => evt.event_type(),
+      SpecChunkEvent::Batch(evt) => evt.event_type(),
+      SpecChunkEvent::Unknown(evt) => evt.event_type(),
+    }
+  }
+}
+
+impl Event for RootChunkEvent {
+  fn event_type(&self) -> &'static str {
+    "RootChunkEvent"
+  }
+}
+
+impl Event for BatchChunkEvent {
+  fn event_type(&self) -> &'static str {
+    "BatchChunkEvent"
+  }
+}
+
+impl Event for UnknownChunkEvent {
+  fn event_type(&self) -> &'static str {
+    "UnknownChunkEvent"
+  }
+}
+
+// TODO: Implement this for an impl Iterator<Item=SpecEvent> rather than requiring a Vec
+impl From<(String, bool, Vec<SpecEvent>)> for SpecChunkEvent {
+  fn from((name, is_root, events): (String, bool, Vec<SpecEvent>)) -> Self {
+    if is_root {
+      Self::Root(RootChunkEvent {
+        id: String::from("root"),
+        name,
+        events,
+      })
+    } else {
+      match BatchChunkEvent::try_from((name, events)) {
+        Ok(batch_chunk) => Self::Batch(batch_chunk),
+        Err((msg, name, events)) => Self::Unknown(UnknownChunkEvent { name, events }),
+      }
```

Comment:
For anything that's not the root chunk (so anything else but specification.json), we attempt to try and parse it as a `BatchChunkEvent` and otherwise store it as `UnknownChunkEvent`.